components/map/events.md

```diff
@@ -304,7 +321,223 @@ The `OnShapeClick` event fires when the user clicks on a shape. Its `EventCallba
+            <MapLayerBubbleSettings>
+                <MapLayerBubbleSettingsStyle>
+                    <MapLayerBubbleSettingsStyleFill Color="#0000ff"></MapLayerBubbleSettingsStyleFill>
+                    <MapLayerBubbleSettingsStyleStroke Color="#000000"></MapLayerBubbleSettingsStyleStroke>
+                </MapLayerBubbleSettingsStyle>
+            </MapLayerBubbleSettings>
+        </MapLayer>
+
+        <MapLayer Type="@MapLayersType.Marker"
+                  Data="@MarkerData1"
+                  LocationField="@nameof(MarkerModel.LatLng)"
+                  TitleField="@nameof(MarkerModel.Title)">
+        </MapLayer>
+    </MapLayers>
+</TelerikMap>
+
+<strong>@EventResult</strong>
+
+@code {
+    private string[] Subdomains { get; set; } = new string[] { "a", "b", "c" };
+    private string UrlTemplate { get; set; } = "https://#= subdomain #.tile.openstreetmap.org/#= zoom #/#= x #/#= y #.png";
+    private string Attribution { get; set; } = "&copy; <a href='https://osm.org/copyright'>OpenStreetMap contributors</a>";
+    private double[] Center { get; set; } = new double[] { 30.268107, -97.744821 };
+    private string EventResult { get; set; }
+
+    private List<MarkerModel> MarkerData1 { get; set; } = new List<MarkerModel>()
+    {
+        new MarkerModel()
+        {
+            LatLng = new double[] { 30.268107, -97.744821 },
+            Title = "Austin, TX"
+        }
+    };
+
+    private List<BubbleModel> BubbleData { get; set; } = new List<BubbleModel>()
+    {
+        new BubbleModel()
+        {
+            LatLng = new double[] { 37.7749, -122.4194 },
+            Revenue = 1000
+        },
+        new BubbleModel()
+        {
+            LatLng = new double[] { 41.8781, -87.6298 },
+            Revenue = 200
+        }
+    };
+
+    private void OnZoomEnd(MapZoomEndEventArgs args)
+    {
+        var zoom = args.Zoom;
+        var center = args.Center;
+        var extent = args.Extent;
+
+        LogToConsole(
+            $"zoom end: zoom level = {zoom}, center coordinates = [{center[0]},{center[1]}]," +
+            $"extent = NW - [{extent[0]}, {extent[1]}]; SE - [{extent[2]}, {extent[3]}]");
+    }
+
+    private void LogToConsole(string text)
+    {
+        EventResult = text;
+    }
+
+    public class MarkerModel
+    {
+        public double[] LatLng { get; set; }
+        public string Title { get; set; }
+    }
+
+    public class BubbleModel
+    {
+        public double[] LatLng { get; set; }
+        public int Revenue { get; set; }
+    }
+}
+````
+
+## OnPanEnd
+
+The `OnPanEnd` event fires when the user has finished moving (panning) the Map. The `OnPanEnd` event handler receives an argument of type `MapPanEndEventArgs` argument, which exposes the following properties:
+
+@[template](/_contentTemplates/common/parameters-table-styles.md#table-layout)
+
+| Property | Type | Description |
+| ---------| ---- | ----------- |
+| `Center` | `double[]` | The lat / lang of the Map's center. |
```

Comment:
Expand the contractions to their complete forms (i.e., latitude and longitude) in both table rows.